src/Symfony/Bridge/Monolog/Processor/DebugProcessor.php

```diff
@@ -33,7 +33,7 @@ public function __construct(RequestStack $requestStack = null)
 
     private function doInvoke(array|LogRecord $record): array|LogRecord
     {
-        $key = $this->requestStack && ($request = $this->requestStack->getCurrentRequest()) ? spl_object_id($request) : '';
+        $key = $this->requestStack && ($request = $this->requestStack->getCurrentRequest(false)) ? spl_object_id($request) : '';
```

Comment:
Everywhere a similar change is made, there is implicit tight coupling to DebugRequestSTack. Eg maybe another implementation is passed and passing false means something else.
To fix this issue, I suggest doing things this way:

```php
    public function __construct(RequestStack $requestStack = null)
    {
        if ($requestStack instanceof DebugRequestStack) {
            $requestStack = $requestStack->withVirtualRequests(); // <- a wither that'd replace the added argument on the methods
        }
        $this->requestStack = $requestStack;
    }
```

Reply:
That would make it impossible to keep DebugRequestStack internal due to cross-component usage right? Too bad given it is not meant to stay

Reply:
an `instanceof` check is not strong coupling - we could drop the class without breaking such snippets

Reply:
Alternatively, instead of replacing the request_stack service, we could create another one that exposes virtual requests, and inject everywhere this behavior is needed. Then, no change in code would be required - only the wiring would know about this. Might be the best option !